src/go/k8s/pkg/resources/configuration/patch.go

```diff
@@ -76,6 +79,24 @@ func ThreeWayMerge(
 	}
 	for k, v := range apply {
 		if oldValue, ok := current[k]; !ok || !PropertiesEqual(log, v, oldValue, schema[k]) {
+			// Value must be type []interface{} if array
+			// https://github.com/redpanda-data/team-kubernetes-internal/issues/11
+			if meta, ok := schema[k]; ok && meta.Type == "array" {
+				value, ok := v.(string)
+				if !ok {
+					log.V(debugLogLevel).Info(fmt.Sprintf("computing patch: expected value to be string, got %T", v), "key", k, "value", v)
+					continue
+				}
+				var a []interface{}
+				if err := yaml.Unmarshal([]byte(value), &a); err != nil {
```

Comment:
I wouldn't go with yaml as yaml is subset of json. Can we switch to json decoder?

Reply:
[yaml is superset of json](https://news.ycombinator.com/item?id=9332362) and we are using yaml in the [rpk](https://github.com/redpanda-data/redpanda/blob/3df51d7711f2ca98918eeb97c6a1f742bc5fcb58/src/go/rpk/pkg/cli/cmd/cluster/config/set.go#L94)

Reply:
:point_up: